File: backend/src/main/kotlin/com/bromles/backend/dto/BookDto.kt
package com.bromles.backend.dto

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer
import java.time.LocalDate

data class BookDto(
    val name: String,
    val author: String,
    val file: ByteArray,
    val tagName: String,
    val category: String,
    @JsonSerialize(using = LocalDateSerializer::class)
    @JsonDeserialize(using = LocalDateDeserializer::class)
    val yearOfPublishing: LocalDate,
)

File: frontend/web/src/app/services/book/dto/book-dto.ts
export class BookDto {
    id?: bigint;
    name?: string;
    author?: string;
    img?: any;
    tagName?:  string;
    category?: string;
    yearOfPublishing?: string;


    public constructor(init?: Partial<BookDto>) {
        Object.assign(this, init);
    }
}
